### ISSUE TITLE
Implement DisneyMaterial

### ISSUE THREAD

**Issue by wahn** [wahn/rs_pbrt#13]
Recently (in July 2017) **Matt Pharr** added the `DisneyMaterial` to the **C++** version of PBRT:

```
commit d9bfba3e9f185894a67563e9c52b55381fcfeef4
Author: Matt Pharr <matt@pharr.org>
Date:   Tue Jul 11 17:07:05 2017 -0700

    Added implementation of Disney BSDF
    
    Specifically, this is the extended model with integrated subsurface
    scattering described in
    http://blog.selfshadow.com/publications/s2015-shading-course/burley/s201
    5_pbs_disney_bsdf_notes.pdf.
```

Let's do the same on the **Rust** side ...

**Comment by wahn** [comment]
Cool. Do you have a **test scene** as well? Otherwise H.P. has one in his repo:

```sh
> grep git ../../.git/config 
	url = https://github.com/hpd/PBRTForMaya.git
> rg disney
pbrt_feature_demo_material_disney.pbrt
53:	MakeNamedMaterial "pbrt_disney"
68:		"string type" ["disney"]
97:		NamedMaterial "pbrt_disney"
154:		NamedMaterial "pbrt_disney"
192:		NamedMaterial "pbrt_disney"
```

**Comment by abusch** [comment]
![pbrt](https://user-images.githubusercontent.com/506344/49511626-c1f11d80-f8df-11e8-86d7-dbb1b89b9035.png)
Hmm... I haven't compared with PBRTv3 yet, but there are some strange artifacts...

**Comment by wahn** [comment]
They have artifacts too:

https://github.com/mmp/pbrt-v3/issues/213

**Comment by wahn** [comment]
Let's investigate the artifacts ...

**Comment by patrickawalz** [comment]
to me it looks like terminator artifact

**Comment by abusch** [comment]
I did a quick render with rustracer for comparison (after quickly hacking support for loading half-float EXR files):
![image](https://user-images.githubusercontent.com/506344/50059931-65e0a000-01e1-11e9-989a-59392c97bdef.png)
It seems close to the PBRTv3 version you posted [there](https://github.com/mmp/pbrt-v3/issues/213). Not sure if I messed something up while porting the code over, or if the problem lies somewhere else...

**Comment by abusch** [comment]
Hmm, yeah, that's possible.
Another thing I've noticed is that the camera angle seem different. Are there some parameters that are not handled yet?

**Comment by wahn** [comment]
The camera angle should be fixed now (after commit 46d5b11939718c2c766fcbfafc1ed62522a0323c). It was the **screenwindow** parameter which wasn't handled ...

**Comment by wahn** [comment]
After commit 61c190f59696f1e19957f8942f8ddd43f55cf279 the Rust version renders exactly the same as the C++ version:

![pbrt_feature_demo_material_disney_rust](https://user-images.githubusercontent.com/1074865/64301754-2b3db400-cf78-11e9-8ada-e174f38e5dec.png)

Closing the issue for now ...
